echo_server.c

```diff
@@ -49,65 +57,109 @@ static int send_request(struct socket *sock, unsigned char *buf, size_t size)
 
     printk(MODULE_NAME ": start send request.\n");
 
+    /* FIXME: what if somehow length of message is zero? */
     length = kernel_sendmsg(sock, &msg, &vec, 1, strlen(buf) - 1);
```

Comment:
`strlen(buf)` can be replaced with `length` returned from kernel_recvmsg

Reply:
Ditto